pkg/azuredx/azuredx.go

```diff
@@ -116,10 +146,49 @@ func NewClient(ctx context.Context, dInfo *backend.DataSourceInstanceSettings) (
 		Scopes:       []string{"https://kusto.kusto.windows.net/.default"},
 	}
 
-	c.Client = conf.Client(ctx)
+	// I hope this correct? The goal is to have a timeout for the
+	// the client that talks to the actual Data explorer API.
+	// One can attach a a variable, oauth2.HTTPClient, to the context of conf.Client(),
+	// but that is the timeout for the token retrieval I believe.
+	// https://github.com/golang/oauth2/issues/206
+	// https://github.com/golang/oauth2/issues/368
+	authClient := oauth2.NewClient(ctx, conf.TokenSource(ctx))
+
+	c.Client = &http.Client{
+		Transport: authClient.Transport,
+		// We add five seconds to the timeout so the client does not timeout before the server.
+		// This is because the QueryTimeout property is used to set the server execution timeout
+		// for queries. The server execution timeout does not apply to retrieving data, so when
+		// a query returns a large amount of data, timeouts will still occur while the data is
+		// being downloaded.
+		// In the future, if we get the timeout value from Grafana's data source proxy setting, we
+		// may have to flip this to subtract time.
+		Timeout: c.dataSourceData.QueryTimeout + 5*time.Second,
+	}
```

Comment:
Looked at the library code and this should work. However, is there any difference to do similar to what you already have? I think `conf.TokenSource(ctx)` refreshes the token if needed so doing this should be the same thing.

```go
c.Client = conf.Client(ctx)
s.Client.Timeout = c.dataSourceData.QueryTimeout + 5*time.Second
```

But in my head shouldn't make a difference only that you would create two HTTP clients instead of one.

Reply:
I did this because the comment on Config.Client says: `// The returned Client and its Transport should not be modified.`, where NewClient I think is for that maybe?

Reply:
Make sense :+1: